app/usecases/characterRepository.go

```diff
@@ -1,27 +1,66 @@
 package usecases
 
 import (
+	"encoding/json"
 	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
 
 	"go-project/app/common"
 	"go-project/app/domain"
 )
 
-const CSVFile string = "/home/amilcar/Documents/Projects/go-project/app/resources/characters.csv"
-const CsvError string = "error while trying to read CSV file"
+const CSVFile string = "/Users/alejandrosanchez/Documents/go_bootcamp/app/resources/characters.csv"
 
 type CharactersRepo struct{}
 
+type Response struct {
+	Results domain.Characters `json:"results"`
+}
 
-func NewCharacterRepo() CharactersRepo{
+func NewCharacterRepo() CharactersRepo {
 	return CharactersRepo{}
 }
 
+func (CharactersRepo) FetchCharacters() (*domain.Characters, error) {
+	response, err := http.Get("https://swapi.dev/api/people")
+	if err != nil {
+		return nil, errors.New(err.Error())
+	}
+
+	responseData, err := ioutil.ReadAll(response.Body)
+
+	if err != nil {
+		return nil, errors.New(err.Error())
+	}
+
+	responseObject := Response{}
+	json.Unmarshal(responseData, &responseObject)
+
+	data, err := common.ReadCsvFile(CSVFile)
+
+	if err != nil {
+		return nil, errors.New(err.Error())
+	}
+
+	lastId := len(data) - 1
+
+	for i := 0; i < len(responseObject.Results); i++ {
```

Comment:
Podemos simplificarlo con un for range

Reply:
Lo estaba intentando con un for range pero no lograba modificar los indices de los structs